plugins/module_utils/managedcluster_addons/cluster_proxy.py

```diff
@@ -37,3 +48,67 @@ def disable_addon(self):
             self.wait,
             self.timeout
         )
+
+    def enable_feature(self):
+        mch = self.get_multi_cluster_hub()
+        changed = False
+        if not self.get_multi_cluster_hub_feature_enablement(mch):
+            # need to update mch
+            self.update_multi_cluster_hub_feature(mch, True)
+            changed = True
+
+        # if self.wait:
+            # waiting for clusterdeployment is not doable
+
+            # cluster_management_addon_api = self.hub_client.resources.get(
+            #     api_version='addon.open-cluster-management.io/v1alpha1',
+            #     kind='ClusterManagementAddOn',
+            # )
+            # try:
+            #     cluster_management_addon_api.get(name=self.addon_name)
+            # except NotFoundError:
+            #     if not self.wait_for_feature_enabled():
+            #         self.module.fail_json(msg=f'timeout waiting for the feature {self.addon_name} to be enabled.')
+        return changed
+
+    def disable_feature(self):
+        mch = self.get_multi_cluster_hub()
+        changed = False
+        if self.get_multi_cluster_hub_feature_enablement(mch):
+            # need to update mch
+            changed = True
+            self.update_multi_cluster_hub_feature(mch, False)
+        return changed
+
+    def update_multi_cluster_hub_feature(self, mch, state=False):
+        mch_api = self.hub_client.resources.get(
+            api_version="operator.open-cluster-management.io/v1",
+            kind="MultiClusterHub",
+        )
+        patch_body = {
+            "apiVersion": "operator.open-cluster-management.io/v1",
+            "kind": "MultiClusterHub",
+            "metadata": {
+                "name": mch.metadata.name,
+                "namespace": mch.metadata.namespace,
+            },
+            "spec": {
+                "enableClusterProxyAddon": state,
+            },
+        }
+        try:
+            mch_api.patch(body=patch_body,
+                          content_type="application/merge-patch+json")
+        except DynamicApiError as e:
+            self.module.fail_json(
+                msg=f'failed to patch MultiClusterHub {mch.metadata.name} in {mch.metadata.namespace} namespace.', err=e)
+
+    def get_multi_cluster_hub_feature_enablement(self, mch):
+        mch_feature_path = ['spec', 'enableClusterProxyAddon']
+        curr = mch
+        for p in mch_feature_path[:-1]:
+            next = curr[p]
+            if next is None:
+                return False
+            curr = next
+        return curr[mch_feature_path[-1]] is True
```

Comment:
next = curr[p]
curr[mch_feature_path[-1]]

If the key doesn't exists, it will throw KeyError exception.   Using get will prevent  KeyError exception
next = curr.get(p)
curr.get(mch_feature_path[-1])

Reply:
good catch 👍 updated ✅